cartoframes/context.py

```diff
@@ -724,9 +734,21 @@ def safe_quotes(text, escape_single_quotes=False):
                      width=size[0],
                      height=size[1],
                      img_html=img_html)
-
-        return IPython.display.HTML(html)
-
+            return IPython.display.HTML(html)
+        elif HAS_MATPLOTLIB:
+            raw_data = mpi.imread(static_url)
+            if ax is None:
+                w,h = size
+                dpi = 1
+                mpl_size = w/dpi, h/dpi # divide by arbitrary dpi to set figure size
+                fig = plt.figure(figsize=mpl_size, dpi=dpi, frameon=False)
+                fig.subplots_adjust(left=0,right=1, top=1, bottom=0)
+                ax = plt.gca()
+            ax.imshow(raw_data)
+            ax.axis('off')
+            return ax
+        else:
+            return IPython.display.HTML(html)
```

Comment:
I was thinking about this a little more. One side effect of the `display.HTML` is that if the underlying dataset changes, the next time someone loads this notebook, they'll get a different image. That's not a good thing if someone is telling a story with a dataset and making changes to it along the way as part of an analysis. To avoid this issue, we should capture a snapshot of the image and have it embedded in the notebook. We can do that with this:

```python
return IPython.display.Image(url=static_url, embed=True, format='png')
```